src/collector/worker.rs

```diff
@@ -0,0 +1,150 @@
+// This software is licensed under a dual license model:
+//
+// GNU Affero General Public License v3 (AGPLv3): You may use, modify, and
+// distribute this software under the terms of the AGPLv3.
+//
+// Elastic License v2 (ELv2): You may also use, modify, and distribute this
+// software under the Elastic License v2, which has specific restrictions.
+//
+// We welcome any commercial collaboration or support. For inquiries
+// regarding the licenses, please contact us at:
+// vectorchord-inquiry@tensorchord.ai
+//
+// Copyright (c) 2025 TensorChord Inc.
+use crate::collector::types::PgRefCell;
+use std::cell::RefMut;
+use std::fs;
+use std::path::Path;
+
+const COLLECTOR_DIR: &str = "pg_vchord_sample";
+const COLLECTOR_VERSION: u32 = 1;
+
+static CONNECTION: PgRefCell<Option<rusqlite::Connection>> =
+    unsafe { PgRefCell::<Option<rusqlite::Connection>>::new(None) };
+
+fn get<'a>() -> Option<RefMut<'a, rusqlite::Connection>> {
+    if unsafe { !pgrx::pg_sys::IsUnderPostmaster } {
+        return None;
+    }
+    let database_oid = unsafe { pgrx::pg_sys::MyDatabaseId.to_u32() };
+    if database_oid == 0 {
+        return None;
+    }
+    let mut connection = CONNECTION.borrow_mut();
+    if connection.is_none() {
```

Comment:
You can extract the code under this branch into an immediately invoked closure, assign the returned `Ok` to connection, and ignore the returned `Err` or print logs.